specutils/fitting/fitmodels.py

```diff
@@ -0,0 +1,507 @@
+        model = [model]
+
+    #
+    # If a single model is passed in then just do that.
+    #
+    fitted_models = []
+
+    for modeli, model_guess in enumerate(model):
+
+        #
+        # Determine the window if it is not None.  There
+        # are several options here:
+        #   window = 4 * u.Angstrom -> Quantity
+        #   window = (4*u.Angstrom, 6*u.Angstrom) -> tuple
+        #   window = (4, 6)*u.Angstrom -> Quantity
+        #
+
+        #
+        #  Determine the window if there is one
+        #
+
+        if window is not None and isinstance(window, list):
+            model_window = window[modeli]
+        elif window is not None:
+            model_window = window
+        else:
+            model_window = None
+
+        fit_model = _fit_lines(spectrum, model_guess, fitter,
+                               exclude_regions, weights, model_window)
+
+        fitted_models.append(fit_model)
+
+    if single_model_in:
+        fitted_models = fitted_models[0]
+
+    return fitted_models
+
+
+def _fit_lines(spectrum, model, fitter=fitting.SimplexLSQFitter(),
+               exclude_regions=None, weights=None, window=None):
+    """
+    Fit the input model (initial conditions) to the spectrum.  Output will be
+    the same model with the parameters set based on the fitting.
+
+    spectrum, model -> model
+
+    Parameters
+    ----------
+    spectrum : Spectrum1D
+        The spectrum object overwhich the equivalent width will be calculated.
+
+    model: `~astropy.modeling.Model`
+        The model or that contain the initial guess.
+
+    exclude_regions : list of `~specutils.SpectralRegion`
+        List of regions to exclude in the fitting.
+
+    weights : list  (NOT IMPLEMENTED YET)
+        List of weights to define importance of fitting regions.
+
+    Returns
+    -------
+    model : Compound model of `~astropy.modeling.Model`
+        A compound model of models with fitted parameters.
+
+    Notes
+    -----
+       * Could add functionality to set the bounds in ``model`` if they are not set.
+       * The models in the list of ``model`` are added together and passed as a
+          compound model to the `~astropy.modeling.fitting.Fitter` class instance.
+
+    """
+
+    #
+    # If we are to exclude certain regions, then remove them.
+    #
+
+    if exclude_regions is not None:
+        spectrum = excise_regions(spectrum, exclude_regions)
+
+    dispersion = spectrum.spectral_axis
+    flux = spectrum.flux
+
+    #
+    # Determine the window if it is not None.  There
+    # are several options here:
+    #   window = 4 * u.Angstrom -> Quantity
+    #   window = (4*u.Angstrom, 6*u.Angstrom) -> tuple
+    #   window = (4, 6)*u.Angstrom -> Quantity
+    #
+
+    #
+    #  Determine the window if there is one
+    #
+
+    # In this case the window defines the area around the center of each model
+    if window is not None and isinstance(window, (float, int)):
+        center = model.mean
```

Comment:
Not all models have the same definition of "center".  My instinct is to right a heuristic function to determine the center - it should work for Gaussian, Lorentzian, and Voigt, and also try some obvious ones like "mean" or "center", but other than that just give up and tell the user "sorry, you've used too weird of a model for me to know what you mean".

That said, a valid response is "won't do in this PR, make an issue to track it for later"!

Reply:
👍